colab/mnist_training.ipynb

```diff
@@ -0,0 +1,860 @@
+          "elapsed": 7759,
+          "user": {
+            "displayName": "Phoenix Meadowlark",
+            "photoUrl": "https://lh3.googleusercontent.com/a-/AOh14Gg1s3aB1Ho9jND1l14tYSizBMuDi8y1thOD0m5z=s64",
+            "userId": "03779344908574432788"
+          }
+        },
+        "outputId": "425f7a7c-f2fb-4f09-a933-81fddb628f0a"
+      },
+      "source": [
+        "# Compile the TrainableDNN module\n",
+        "compiled_model = module_utils.IreeCompiledModule.create_from_class(\n",
+        "    TrainableDNN, \n",
+        "    module_utils.BackendInfo(backend_choice), \n",
+        "    exported_names=exported_names)"
+      ],
+      "execution_count": 12,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "INFO:tensorflow:Assets written to: /tmp/tmprccz86q3.sm/assets\n"
+          ],
+          "name": "stdout"
+        },
+        {
+          "output_type": "stream",
+          "text": [
+            "I0114 19:46:22.975932 140420144105344 builder_impl.py:775] Assets written to: /tmp/tmprccz86q3.sm/assets\n",
+            "2021-01-14 19:46:23.145960: I external/org_tensorflow/tensorflow/cc/saved_model/bundle_v2.cc:32] Reading SavedModel from: /tmp/tmprccz86q3.sm\n",
+            "2021-01-14 19:46:23.147528: I external/org_tensorflow/tensorflow/cc/saved_model/bundle_v2.cc:55] Reading SavedModel debug info (if present) from: /tmp/tmprccz86q3.sm\n",
+            "2021-01-14 19:46:23.175245: W external/org_tensorflow/tensorflow/compiler/mlir/tensorflow/translate/import_model.cc:1795] Unmodelled op type `StaticRegexFullMatch` is not stateful but will be treated as such conservatively\n",
+            "2021-01-14 19:46:23.175427: W external/org_tensorflow/tensorflow/compiler/mlir/tensorflow/translate/import_model.cc:1795] Unmodelled op type `SaveV2` is stateful but effects not modelled\n",
+            "Created IREE driver dylib: <pyiree.rt.binding.HalDriver object at 0x7fb5afc56d18>\n",
+            "SystemContext driver=<pyiree.rt.binding.HalDriver object at 0x7fb5afc56d18>\n"
+          ],
+          "name": "stderr"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "DtOS_rd_BP5V"
+      },
+      "source": [
+        "## 6. Train the Compiled Model on MNIST"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "2MVsxIy5BdqF"
+      },
+      "source": [
+        "Note: At the time of writing IREE is about 20x slower on desktop CPU than TensorFlow, so there's no advantage to using it for training in this situation.\n",
+        "\n",
+        "Since the compiled model is portable however, this demonstartes that IREE can be used for training on a mobile device (e.g. a Pixel). In that setting IREE has a ~1000 fold binary size advantage over the current TensorFlow solution (which is to use the now-deprecated TF Mobile, as TFLite does not support training at this time)."
```

Comment:
I'd leave out the specific performance note here, or reword to be a bit more charitable :P. IREE's CPU performance is still being improved, but IREE plans to be "just" a low level system for compiling/executing kernels. Training in full TF comes with many other advantages, like being able to distribute work across many machines and interop with monitoring services.

The rest is good info though.

One suggestion:

```suggestion
        "This compiled model is portable, demonstrating that IREE can be used for training on a mobile device. On mobile, IREE has a ~1000 fold binary size advantage over the current TensorFlow solution (which is to use the now-deprecated TF Mobile, as TFLite does not support training at this time)."
```